web_src/js/markup/anchors.js

```diff
@@ -31,5 +33,23 @@ export function initMarkupAnchors() {
     heading.prepend(a);
   }
 
+  // handle user-defined `name` anchors like `[Link](#link)` linking to `<a name="link"></a>Link`
+  for (const a of document.querySelectorAll('.markup a[href^="#"]')) {
+    const href = a.getAttribute('href');
+    if (!href.startsWith('#user-content-')) continue;
+    const originalId = href.replace(/^#user-content-/, '');
+    a.setAttribute('href', `#${encodeURIComponent(originalId)}`);
+    const anchorDest = document.getElementsByName(originalId);
+    if (anchorDest.length === 1) {
+      a.addEventListener('click', () => {
+        anchorDest[0].scrollIntoView();
+      });
+    } else {
+      a.addEventListener('click', (e) => {
+        scrollToAnchor(e.currentTarget.getAttribute('href'), false);
+      });
+    }
```

Comment:
```suggestion
```


I see no behaviour difference when removing this block because once the `href` on the source link is corrected, it will work natively. Would you agree?

Reply:
I agree that the first branch of the code you suggested to be removed is unnecessary since the link will work natively, as you suggested. However, by removing the second branch, the markup link targets that do not use `<a name="link"></a>` will break, because we have removed "user-content-" in the links, so it would not link to the corresponding id's. I believe this event listener would solve that issue. 

I have tried my suggestion in an example, and it worked as expected. What do you think about my proposal?

Reply:
You are right, please do like this:

```js
    if (document.getElementsByName(originalId).length !== 1) {
      a.addEventListener('click', (e) => {
        scrollToAnchor(e.currentTarget.getAttribute('href'), false);
      });
    }